ome_zarr/writer.py

```diff
@@ -122,6 +122,23 @@ def _validate_plate_rows_columns(
     return validated_list
 
 
+def _validate_datasets(datasets: List[dict]) -> List[Dict]:
+
+    validated_datasets = []
+    if datasets is None or len(datasets) == 0:
+        raise ValueError("Empty datasets list")
+    for dataset in datasets:
+        if isinstance(dataset, str):
```

Comment:
I assume this block can be completely removed now and `datasets == str` should be handled by the `ValueError` below

Reply:
That didn't quite work because we still support `List[str]` list of paths for `< 0.4`. Lots of tests failed when I removed it.

Reply:
Given that this API was introduced in  https://github.com/ome/ome-zarr-py/pull/149 and never made available in a public release, I would be personally fine with completely dropping this form and updating all the failing tests to use the new datasets form. The alternative would be to add some form of check on the format version here and fail hard if `fmt.version` is not 0.3 or lower.

I think the biggest issue I have reading the code is that it looks like there is a path for using this API and creating a `0.4` version of the `multiscales` with `datasets` only composed of `path` which is invalid. I can look into that in a follow-up PR.